authn.js

```diff
@@ -73,12 +73,17 @@ function setup(app) {
           Username: profile.username
         }).promise();
 
-        const groups = response.Groups
-          .map((g) => g.GroupName)
-          .filter((g) => sources.has(g));
+        const groups = response.Groups.map((g) => g.GroupName);
 
         // All users are ok, as we control the entire user pool.
-        return done(null, {...profile, groups});
+        const user = {...profile, groups};
+
+        user.visibleSources = Array.from(sources.values())
+          .filter(source => source.visibleToUser(user))
+          .map(source => source._name)
+          .filter(source => groups.includes(source));
```

Comment:
Is the purpose of this last line to filter out the non-group sources (e.g. `CoreSource`, `CoreStagingSource`, and `CommunitySource`)?

If so, then I think there are more future-proof ways of doing this which delegates that decision to the source itself and avoids deepening the existing hardcoded assumption/conflation that the group name matches the source name exactly.

I don't think :point_up: needs to block merge, but I'll think about it some more and maybe push a commit up.

Reply:
Yes, it's to remove the groups/tiles that appear as `core`, `staging`, and `community` on nextstrain.org.

Reply:
I'm curious to know what a better solution would be.

Reply:
> I'm curious to know what a better solution would be.

Not entirely sure! :-) The equating of group names with source names in this part of the code raises red flags for me, but I think avoiding that here would require broader changes than I think are worth implementing right now. So I would merge this as-is and we can revisit later during other work on Groups if it arises again.